Fixed 403 error when page refreshes

MarekB01 opened the pull request:
https://github.com/BroadleafCommerce/QA/issues/4723

I changed the logic to use extension handler and changed a validation logic so it can use parent id of product

MarekB01 committed:
BroadleafCommerce/QA#4723_fixed_403_error

I changed the logic to set property value from SectionCrumb id, so the check will not fail anyway.

danielcolgrove commented on admin/broadleaf-open-admin-platform/src/main/java/org/broadleafcommerce/openadmin/server/service/persistence/PersistenceManagerImpl.java:
```diff
@@ -421,8 +421,8 @@ protected void postFetchValidation(PersistencePackage persistencePackage, Persis
                         .findFirst();
-                    if (property != null && sectionCrumbOptional.isPresent()
-                        && !property.getValue().equals(sectionCrumbOptional.get().getSectionId())) {
+                    property.setValue(sectionCrumbOptional.get().getSectionId());
```
You can't change the property.  The solution should be similar to https://github.com/BroadleafCommerce/Enterprise/pull/1452 in that we need to take the property id and find its parent.  The issue here is that this code is in framework and the logic we need is in Enterprise.  So we will likely need an ExtensionHandler to call into Enterprise to get the parent id.

MarekB01 committed:
BroadleafCommerce/QA#4723_fixed_403_error

I provided ExtensionHandling logic to check if the property value is overriden and validation logic works correct

KatrukOV commented on admin/broadleaf-open-admin-platform/src/main/java/org/broadleafcommerce/openadmin/handler/ParentIdServiceExtensionManager.java:
```diff
@@ -0,0 +1,28 @@
+@Service("blParentIdServiceExtensionManager")
+public class ParentIdServiceExtensionManager extends ExtensionManager<BasicParentIdServiceExtensionHandler> {
+
+    public ParentIdServiceExtensionManager(){super(BasicParentIdServiceExtensionHandler.class);}
```
use code formatting

KatrukOV commented on admin/broadleaf-open-admin-platform/src/main/java/org/broadleafcommerce/openadmin/server/service/persistence/PersistenceManagerImpl.java:
```diff
@@ -422,8 +429,16 @@ protected void postFetchValidation(PersistencePackage persistencePackage, Persis
+                            && !property.getValue().equals(sectionCrumbOptional.get().getSectionId())) {
+                        ExtensionResultHolder<String> extensionResultHolder = new ExtensionResultHolder<String>();
+                        if (extensionManager.getProxy().checkParentId(property.getValue(),foreignKey.getForeignKeyClass(), extensionResultHolder) != ExtensionResultStatusType.NOT_HANDLED ){
+                           String parentID = extensionResultHolder.getResult();
```
AbstractParentIdServiceExtensionHandler.checkParentId() always return ExtensionResultStatusType.NOT_HANDLED;

Is it correct?

danielcolgrove committed:
Moved manager and handler to the common package.  Removed the abstract handler as it isn't needed.  Renames to EntityParentIdServiceExtension...

danielcolgrove committed:
The logic was changed and incorrectly validating when we didn't have sectioncrumb data.
